ledger/ledger-common/src/main/java/net/corda/v5/ledger/common/transaction/TransactionSignatureVerificationService.java

```diff
@@ -0,0 +1,54 @@
+package net.corda.v5.ledger.common.transaction;
+
+import net.corda.v5.application.crypto.DigitalSignatureAndMetadata;
+import net.corda.v5.application.crypto.DigitalSignatureVerificationService;
+import net.corda.v5.base.annotations.DoNotImplement;
+import net.corda.v5.base.annotations.Suspendable;
+import net.corda.v5.crypto.SecureHash;
+import net.corda.v5.crypto.merkle.MerkleProof;
+import org.jetbrains.annotations.NotNull;
+
+import java.security.PublicKey;
+import java.util.List;
+
+/**
+ * TransactionSignatureVerificationService can be used to verify transaction signatures.
+ * It supports both single and batch signatures.
+ * It can be used in both flows and contracts.
+ */
+@DoNotImplement
+public interface TransactionSignatureVerificationService {
+    /**
+     * Verifies a signature against a transaction.
+     * The underlying verification service signals the verification failures with different exceptions.
+     * {@link DigitalSignatureVerificationService}
+     *
+     * @param transaction           The original transaction.
```

Comment:
This method (verifySignature against TransactionWithMetadata) uses the metadata only to cross-match the batch settings against the digital signature's batch settings.
The verification itself works without the transaction's metadata.
We could deprecate this function if we do not need this extra verification logic.
But since that would be an API breaking change, it needs to stay.